afex_integration/views/account_view.xml

```diff
@@ -35,6 +39,9 @@
         <div name="amount_div" position="after">
             <field name="is_afex" invisible="1"/>
             <field name='passed_currency_id' invisible='1'/>
+            <field name="aud_currency_id" invisible="1"/>
+            <field name="afex_direct_debit_journal_id" invisible="1"/>
+            <field name="afex_direct_debit" attrs="{'invisible': ['|', '|', ('is_afex', '=', False), ('aud_currency_id', '=', False), ('afex_direct_debit_journal_id', '=', False)]}"/>
```

Comment:
remove currency from consideration - invisible if not afex and not afex_drect_debit_journal_id is sufficient.